core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3227,6 +3240,28 @@ public void validateIntervalQualifier(SqlIntervalQualifier qualifier) {
     }
   }
 
+  @Override public void validateCreateFunctionSqlForm(
+      SqlCreateFunctionSqlForm createFunction) {
+    Preconditions.checkArgument(createFunction != null);
```

Comment:
What would happen if this method were called twice with the same function signature (if the user accidentally defined the same function twice, or wanted to define it and then overwrite its behavior)?

Reply:
I just tried it and it results in the "duplicate" call being added to the schema as well. Behavior wise the function which was added first gets chosen when matching. I'm not sure how much of an issue this is as usually a "DROP" call comes before redefining a structure right?

Reply:
Though it should be possible to add the ability to "overwrite" entries that have an already matching signature.

Reply:
Not sure if this is an assumption we can rely on - @dasch-1, what is the expected behavior if the same function gets created twice?

Reply:
I think it makes sense to overwrite a function of the same signature if it is added again.

Reply:
I seem to remember you may get an error if you try to CREATE the same thing more than once in SQL.

Reply:
Just to clarify, we're only talking about a CREATE FUNCTION not a REPLACE FUNCTION, right?

Reply:
I believe so.

Reply:
Since two subsequent CREATE (but not necessarily CREATE OR REPLACE, or REPLACE) calls should be illegal, I think we can handle this in any reasonable way we want. Overriding the previous definition makes sense. Or throwing an error makes sense too.

But we probably don't want both function signatures to exist simultaneously in the catalog with only one that's accessible.

Reply:
I decided to go with the overwriting approach and added a test to demonstrate this.